client/domain/PrereqCheckFailureObjectFactory.js

```diff
@@ -197,10 +241,14 @@ tie.factory('PrereqCheckFailureObjectFactory', [
      *    unsupported libraries the user tried to import
      * @param {string} starterCode Should have the original code and function
      *    headers the user is given before they begin adding their own code.
+     * @param {string} wrongLangName Name of the error that corresponds to the
+     *    error information in WRONG_LANGUAGE_ERRORS
      * @returns {PrereqCheckFailure}
      */
-    PrereqCheckFailure.create = function(type, badImports, starterCode) {
-      return new PrereqCheckFailure(type, badImports, starterCode);
+    PrereqCheckFailure.create = function(
+        type, badImports, starterCode, wrongLangName) {
+      var name = (typeof wrongLangName === 'undefined') ? null : wrongLangName;
```

Comment:
A common "fallback to default value" syntax in JS is: `var blah = wrongLangName || null;`